### ISSUE TITLE
[Question] is it possible to disable code level annotations?

### ISSUE THREAD

**Issue by philippe-boyd-3r** [rust-lang/rust-clippy#5836]
When running `cargo clippy`, is it possible to disable code level annotations?

For instance, in the code (`src/lib.rs`) I have:
```rust
#![allow(warnings)]

// Code follows...
```

However, when calling (from a CI for instance):
```bash
cargo clippy -- -D warnings
```
code level annotations will take precedence over the ones mentioned on the command line meaning warnings will be allowed.

We don't want the code to be able to bypass CI checks, in this case, clippy lints.

**Comment by flip1995** [comment]
There is the `forbid` level for lints. So if you call `cargo clippy -- -Fclippy::all` (or `-Fwarnings`) every time a lint level is changed in the code will error. [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e6b59243af8c9cc94cf40103312c7311)

I wouldn't `allow(warnings)` or `allow(clippy::all)` during development though. What Clippy does for this is to use a `deny-warnings` feature, which is then turned on in CI. That way, you still get warnings locally, but CI will error if there are warnings:

https://github.com/rust-lang/rust-clippy/blob/8c83d5f484950d86630f467d32efcd585de2d775/Cargo.toml#L56-L57

https://github.com/rust-lang/rust-clippy/blob/8c83d5f484950d86630f467d32efcd585de2d775/src/main.rs#L1

https://github.com/rust-lang/rust-clippy/blob/8c83d5f484950d86630f467d32efcd585de2d775/.github/workflows/clippy.yml#L68-L69

**Comment by philippe-boyd-3r** [comment]
TIL about the forbid `-f` flag. The more you know... thanks!
PS: and yes of course, the `allow(warnings)` was just as an example.
